src/components/docs/LeftSidebar/Section.tsx

```diff
@@ -26,71 +33,92 @@ export const Section = ({ text, items, collapsible, link }: SectionProps) => {
 
   const router = useRouter();
 
-  const toLink = () => {
-    if (link) {
-      router.push(link);
+  useEffect(() => {
+    if (link && router.asPath) {
+      if (!router.asPath.includes(link) === false) {
+        setCollapsed(false);
+      }
     }
-  };
+  }, [link, router.asPath]);
 
-  const sectionIsCollapsable = useMemo(() => {
-    if (!link) {
-      if (items.length === 0) return false;
-      return collapsible;
-    } else {
-      return false;
-    }
-  }, [link, collapsible, items]);
+  if (isHeader) {
+    return (
+      <h1 className="mb-3 mt-4 flex-1 text-sm font-semibold uppercase text-black dark:text-instillGrey05">
+        {t(text)}
+      </h1>
+    );
+  }
 
   return (
-    <section
-      className={cn(
-        "flex w-full flex-col pt-2",
-        sectionIsCollapsable ? (collapsed ? "pb-2" : "pb-6") : "pb-2"
-      )}
-    >
+    <section className={cn("flex w-full flex-col pb-1 pt-1")}>
       <div
-        onClick={link ? toLink : toggle}
         role="button"
         className={cn(
-          "flex flex-row",
-          sectionIsCollapsable ? { "mb-4": !collapsed } : ""
+          "flex flex-row px-1 py-1.5 hover:rounded hover:bg-instillGrey05 dark:text-instillGrey05 dark:hover:bg-instillGrey80",
+          collapsible ? { "mb-1": !collapsed } : "",
```

Comment:
@iamnamananand996 This can be written as
```suggestion
         { "mb-1": !collapsed }: collapsible,
```

Reply:
And I will suggest that we put the string at top and the logic at bottom to make it more readable